src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -401,4 +401,22 @@ private StatsForInstructorDashboardDTO populateCommonStatistics(Exercise exercis
 
         return ResponseUtil.wrapOrNotFound(Optional.of(exercise));
     }
+
+    /**
+     * GET /exercises/:exerciseId/toggle-second-correction
+     *
+     * @param exerciseId the exerciseId of the exercise to get the repos from
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @GetMapping(value = "/exercises/{exerciseId}/toggle-second-correction")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Boolean> toggleSecondCorrectionEnabled(@PathVariable Long exerciseId) {
+        log.debug("toggleSecondCorrectionEnabled for exercise with id:" + exerciseId);
+        Exercise exercise = exerciseService.findOne(exerciseId);
```

Comment:
You should maybe send a Not Found if the exercise with the id does not exist.

Reply:
Done 👍